ddp-workspace/projects/ddp-testboston/src/app/components/header/header.component.ts

```diff
@@ -21,10 +25,36 @@ export class HeaderComponent {
     return this.session.isAuthenticatedSession();
   }
 
+  public openMenu(): void {
+    this.isMenuOpened = true;
+    this.window.nativeWindow.requestAnimationFrame(() => {
+      this.renderer.addClass(this.menu.nativeElement, 'menu_visible');
+    });
+    this.window.nativeWindow.requestAnimationFrame(() => {
+      this.renderer.addClass(this.overlay.nativeElement, 'overlay_visible');
+    });
+  }
+
+  public closeMenu(): void {
+    if (this.isMenuOpened) {
+      this.isMenuOpened = false;
+      this.window.nativeWindow.requestAnimationFrame(() => {
+        this.renderer.removeClass(this.overlay.nativeElement, 'overlay_visible');
+      });
+      this.window.nativeWindow.requestAnimationFrame(() => {
+        this.renderer.removeClass(this.menu.nativeElement, 'menu_visible');
+      });
+    }
+  }
+
   @HostListener('window: scroll') public onWindowScroll(): void {
     const scrolledPixels = this.window.nativeWindow.pageYOffset
       || this.document.documentElement.scrollTop
       || this.document.body.scrollTop || 0;
     this.isPageScrolled = !!scrolledPixels;
   }
+
+  @HostListener('window: resize') public onWindowResize(): void {
+    this.closeMenu();
+  }
```

Comment:
Interesting little tidbit. How do resizes work on mobile?

Reply:
It isn't for mobiles, it's for desktop users who resize the browser window.
If you remove this code and make the following actions you will see weird behavior:
1. Reduce the size of the browser window so that the Menu button appears
2. Open menu
3. Increase the size of the browser window so that the Menu button is hidden
4. You will see the opened menu and desktop header 
This method prevents the described behavior.